### DISCUSSION TITLE
No such mapping after upgraded to 12.31.0

### DISCUSSION THREAD

**Discussion by tranbinh1604** [LazyVim/LazyVim#4098] [Q&A]
Hello, 

I'm getting an error about key mapping after upgrading to 12.31.0:

```
vim/keymap.lua:0: E31: No such mapping

# stacktrace:
  - vim/keymap.lua:0 _in_ **del**
  - lua/config/nomap.lua:8
  - lua/config/keymaps.lua:1
  - /LazyVim/lua/lazyvim/config/init.lua:247
  - /LazyVim/lua/lazyvim/config/init.lua:246 _in_ **_load**
  - /LazyVim/lua/lazyvim/config/init.lua:257 _in_ **load**
  - /LazyVim/lua/lazyvim/config/init.lua:183
```

My configuration is like this: 
- `lua/config/nomap.lua` which will disable some key binding which I don't want to use, it will required in `lua/config/keymaps.lua` as below:
```
require("config.nomap") -- the error from this line 

local map = vim.keymap.set

map("n", ";", ":", { desc = "cmd enter command mode" })
``` 

- Here is the content of `nomap.lua`:

```
local del = vim.keymap.del

-- the error starts from here
del("n", "<leader>ww") 
del("n", "<leader>wd") 
del("n", "<leader>w-") 
del("n", "<leader>w|") 
del("n", "<leader>wm") 
-- the error ends here 

-- all below still work as well
del({ "i", "x", "n", "s" }, "<C-s>") -- save file
del("n", "<leader>-") -- split win horizontally
del("n", "<leader>|") -- split win vertically
del("n", "<leader>`") -- switch to other buffer

del("n", "<leader><tab>l") -- goto last tab
del("n", "<leader><tab>o") -- close other tabs
del("n", "<leader><tab>f") -- first tab
del("n", "<leader><tab><tab>") -- new tab
del("n", "<leader><tab>]") -- next tab
del("n", "<leader><tab>d") -- delete tab
del("n", "<leader><tab>[") -- previous tab

del("n", "<S-h>")
del("n", "<S-l>")
```

In version 12.30.0 and earlier, everything worked fine, but after 12.31.0, I'm getting the error above. I tried to check with `:verbose nmap <leader>ww` and got:

```
14:44:42 msg_show   verbose nmap <leader>ww n  <Space>w    * <C-W>
                 Windows
	Last set from Lua (run Nvim with -V1 for more details)
```
This means there is no mapping for this, which explains why the "no such mapping" error occurred. However, when I press `<leader>w`, I get all keymaps exactly the same as the ones I want to disable in `nomap.lua` file, 

<img width="1906" alt="image" src="https://github.com/user-attachments/assets/01fadc77-6c1d-44c4-936a-49b6ef646683">

My `which-key.lua` also quite simple:

```
return {
  "folke/which-key.nvim",
  opts = {
    plugins = { spelling = true },
    specs = {
      {
        { "<leader>h", group = "harpoon/hunks", mode = { "n", "v" } },
      },
    },
  },
}
```

is there anyway to fix this? or how to stay with `12.30.0`, I changed `version = "v12.30.0", -- always use the latest git commit
` but the update notification alway appear and quite annoying even when the exactly version already set.

**Comment by dpetka2001** [comment]
Just remove the offending keymaps. Those are provided by which-key `expand` property now, that's why they don't exist as regular keymaps anymore, but rather expand to the Neovim default behavior of `<C-w>`.

**Reply by tranbinh1604** [reply]
can you help me how to do that? im actually quite new with nvim at all :(

**Reply by dpetka2001** [reply]
Just remove 
```lua
del("n", "<leader>ww") 
del("n", "<leader>wd") 
del("n", "<leader>w-") 
del("n", "<leader>w|") 
del("n", "<leader>wm") 
```
from your `nomap.lua`.

**Reply by tranbinh1604** [reply]
well, remove them can solve problem, but im having other keymaps which will use the same above but can not overwrite the default ones.

for example I want to use `<leader>ww` to save the current file instead of `<C-s>`, but unable to overwrite the default one `switch window`, hence Im trying to find a way to disable keymaps which I dont use.

**Reply by dpetka2001** [reply]
I don't know if you can directly overwrite the defaults. Your best bet would be to either remap it to some other key that is not mapped yet or move the default `<leader>w` to another mapping (like `<leader>W` for example) and then be able to use `<leader>w` for whatever you want. Maybe the maintainer will have a better idea of how to achieve what you want, if you'd like to ask him.

**Reply by tranbinh1604** [reply]
hi @dpetka2001 , thanks you, im also trying to research which-key to findout is there anyway to make it work, normally in version `12.30.0` the keymaps not provide by which-key, quite easy to fix it, but now I dont know where to start at all :( 

is there anyway to keep the version in `12.30.0` without lazyvim update notification?

**Reply by dpetka2001** [reply]
So will you stay in LazyVim 12.30.0 for ever? That doesn't make sense. The correct way is to find a solution. Try the following
```lua
  {
    "folke/which-key.nvim",
    opts = {
      spec = {
        { "<leader>w", group = "Save" },
        { "<leader>ww", "<cmd>w<cr><esc>", desc = "Save Buffer" },
        {
          "<leader>W",
          proxy = "<c-w>",
          group = "windows",
        },
      },
    },
  },
```
That moves all windows operations under `<leader>W` and you are free to use `<leader>w` for whatever you want. Again, I'm not sure if this is the best solution. You might want to ask the maintainer if there's a better way to achieve what you want. 

@folke Sorry for pinging you, but your input would be greatly appreciated regarding this matter. I'm also not that familiar with all `which-key` newest things, so that was the best I could come up with. Is there a way to change the default `<leader>ww` to what a user wants? Neither setting it in `keymaps.lua` via `vim.keymap.set` nor trying to define it under which-key `opts.spec` seems to do what the user wants. Of course I might be missing something obvious as well.

**Reply by folke** [reply]
Your approach works, but I also just pushed a wk change so that custom mappings always override those from proxy mappings.

**Reply by dpetka2001** [reply]
Cool, tnx a lot for letting us know :smile:
